hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -6131,15 +6131,17 @@ void releaseBackupNode(NamenodeRegistration registration)
   static class CorruptFileBlockInfo {
     final String path;
     final Block block;
+    private final int replication;
     
-    public CorruptFileBlockInfo(String p, Block b) {
+    CorruptFileBlockInfo(String p, Block b, int r) {
       path = p;
       block = b;
+      replication = r;
     }
     
     @Override
     public String toString() {
-      return block.getBlockName() + "\t" + path;
+      return block.getBlockName() + "\t" + replication + "\t" + path;
```

Comment:
I am a little worried about the compatibility while upstream dependency with this string but we change the format here.